src/status_im2/contexts/communities/discover/view.cljs

```diff
@@ -104,47 +118,126 @@
+                                                       [community/community-actions community])}])}
+            (merge community
+                   (get mock-community-item-data :data))])]))
+    (if communities communities communities-ids))])
+
+
+(defn communities-list-component-fn
+  [selected-tab view-type]
+  (let [ids-by-user-involvement (rf/sub [:communities/community-ids-by-user-involvement])
+        all-communities         (rf/sub [:communities/sorted-communities])
+        tab                     @selected-tab]
+    [rn/view {:style {:flex 1}}
+     (case tab
+       :all
+       (other-communities-list {:communities all-communities
+                                :view-type   view-type})
+
+       :open
+       (other-communities-list {:communities-ids (:open ids-by-user-involvement)
+                                :view-type       view-type})
+
+       :gated
+       (other-communities-list {:communities-ids (:gated ids-by-user-involvement)
+                                :view-type       view-type})
+
+       [quo/information-box
+        {:type :error
+         :icon :i/info}
+        (i18n/label :t/error)])]))
+
+(def discover-communities-list-component (memoize communities-list-component-fn))
+
+(defn render-page-content
+  [selected-tab view-type]
+  (fn []
+    [rn/view {:padding-horizontal 20}
+     [discover-communities-list-component selected-tab view-type]]))
+
+(defn render-sticky-header
+  [selected-tab]
+  (fn [scroll-height]
+    (when (> scroll-height 348)
+      [react/blur-view
+       {:blur-amount   32
+        :blur-type     :xlight
+        :overlay-color (if platform/ios? colors/white-opa-70 :transparent)
+        :style         style/blur-tabs-header}
+       [discover-communities-segments selected-tab]])))
+
+(defn discover-screen-content
+  [featured-communities]
+  (let [view-type (reagent/atom :card-view)
+        selected-tab (reagent/atom :all)
+        featured-communities-count (count featured-communities)
+        scroll-component (scroll-page/scroll-page
+                          {:background-color  (colors/theme-colors
+                                               colors/white
+                                               colors/neutral-95)
+                           :content-container (fn []
+                                                [rn/view {:margin-top 112}
+                                                 [discover-communities-header
+                                                  {:featured-communities-count featured-communities-count
+                                                   :featured-communities       featured-communities
+                                                   :view-type                  @view-type
+                                                   :selected-tab               selected-tab}]])
+                           :name              (i18n/label :t/discover-communities)})]
+    (fn []
```

Comment:
I think we shouldn't use form-2 components like this, but please, correct me if I'm wrong. `scroll-component` is declared outside the renderer function, and it relies on `featured-communities` argument. If `featured-communities` change, the `scroll-component` will point to the outdated value. It doesn't seem correct to me, hence `scroll-component` should be part of renderer, not outside of it.

Once you move `scroll-component` to the renderer, you'll need to repeat the argument. Even if the code works now, Reagent suggests we pay close attention to this because it leads to subtle bugs, hard to uncover ones.

For more info, see the rookie mistake section on https://github.com/reagent-project/reagent/blob/6d2b0d2a1dbbdaca96840e2e09c5d50edcdced94/doc/CreatingReagentComponents.md#form-2--a-function-returning-a-function